src/ruby/puppetserver-lib/puppet/server/logging.rb

```diff
@@ -0,0 +1,26 @@
+require 'puppet/server/log_collector'
+
+module Puppet
+  module Server
+    module Logging
+
+      def self.capture_logs(level, &block)
+        old_log_level = Puppet.settings[:log_level]
+        Puppet.settings[:log_level] = level if level
```

Comment:
What would an invocation of `Logging.capture_logs(false) do .. end` mean?

I would guess that `level` is more of a logging level override and might make more sense as an optional param if we want callers to use Puppet Server's default logging level.

Reply:
That makes sense.

Reply:
Supplying the `nil` here as a default (not overriding puppetserver's log level) is functionally what this does already, since if the option wasn't present in the request, nil will be passed. We don't have any places in code where `level` is not passed in. Did you have something different in mind than just `capture_logs(level = nil, &block)`?

Reply:
not really, though I might change the name of the variable to something conveying that the level is an override of the existing process logging level. or a comment/doc to that effect with valid values.